src/ray/raylet/agent_manager.h

```diff
@@ -32,96 +30,56 @@ typedef std::function<std::shared_ptr<boost::asio::deadline_timer>(std::function
+    if (options_.agent_name.empty()) {
+      RAY_LOG(FATAL) << "AgentManager agent_name must not be empty.";
+    }
+    if (options_.agent_commands.empty()) {
+      RAY_LOG(FATAL) << "AgentManager agent_commands must not be empty.";
+    }
     if (start_agent) {
       StartAgent();
     }
   }
-
-  void HandleRegisterAgent(rpc::RegisterAgentRequest request,
-                           rpc::RegisterAgentReply *reply,
-                           rpc::SendReplyCallback send_reply_callback) override;
-
-  /// Request agent to increase the runtime env reference. This API is not idempotent.
-  /// \param[in] job_id The job id which the runtime env belongs to.
-  /// \param[in] serialized_runtime_env The serialized runtime environment.
-  /// \param[in] serialized_allocated_resource_instances The serialized allocated resource
-  /// instances.
-  /// \param[in] callback The callback function.
-  virtual void GetOrCreateRuntimeEnv(
-      const JobID &job_id,
-      const std::string &serialized_runtime_env,
-      const rpc::RuntimeEnvConfig &runtime_env_config,
-      const std::string &serialized_allocated_resource_instances,
-      GetOrCreateRuntimeEnvCallback callback);
-
-  /// Request agent to decrease the runtime env reference. This API is not idempotent.
-  /// \param[in] serialized_runtime_env The serialized runtime environment.
-  /// \param[in] callback The callback function.
-  virtual void DeleteRuntimeEnvIfPossible(const std::string &serialized_runtime_env,
-                                          DeleteRuntimeEnvIfPossibleCallback callback);
+  ~AgentManager();
 
  private:
   void StartAgent();
 
  private:
   Options options_;
-  pid_t reported_agent_id_ = 0;
-  int reported_agent_port_ = 0;
-  /// Whether or not we intend to start the agent.  This is false if we
-  /// are missing Ray Dashboard dependencies, for example.
-  bool should_start_agent_ = true;
-  std::string reported_agent_ip_address_;
+  Process process_;
   DelayExecutorFn delay_executor_;
-  RuntimeEnvAgentClientFactoryFn runtime_env_agent_client_factory_;
-  std::shared_ptr<rpc::RuntimeEnvAgentClientInterface> runtime_env_agent_client_;
-  /// When the grpc port of agent is invalid, set this flag to indicate that agent client
-  /// is disable.
-  bool disable_agent_client_ = false;
-};
-
-class DefaultAgentManagerServiceHandler : public rpc::AgentManagerServiceHandler {
- public:
-  explicit DefaultAgentManagerServiceHandler(std::shared_ptr<AgentManager> &delegate)
-      : delegate_(delegate) {}
-
-  void HandleRegisterAgent(rpc::RegisterAgentRequest request,
-                           rpc::RegisterAgentReply *reply,
-                           rpc::SendReplyCallback send_reply_callback) override {
-    RAY_CHECK(delegate_ != nullptr);
-    delegate_->HandleRegisterAgent(request, reply, send_reply_callback);
-  }
-
- private:
-  std::shared_ptr<AgentManager> &delegate_;
+  bool kill_raylet_if_process_exits_;
```

Comment:
Use atomic bool instead? (used by 2 threads)

Reply:
It should be fine since the 2 threads are synchronized on process kill/wait. But yeah, no harm to add an atomic read. Added.

Reply:
Also renamed to fate_shares_.